libs/frontend/ui-build/src/components/ComponentTree.vue

```diff
@@ -44,25 +37,27 @@ const treeDepth = computed(() => {
   if (!root || !expandedCmp) {
     return 0
   }
-  let depth = 0
+
   // Tree iteration
-  const stack = [makeNode(root, 0)]
-  while (stack.length > 0) {
-    const node = stack.pop()
-    if (node && expandedCmp[node.cmp.id]) {
-      if (node.depth > depth) {
-        depth = node.depth
-      }
-      if (node.cmp.children) {
-        stack.push(...node.cmp.children.map((c) => makeNode(c, depth + 1)))
+  let depth = 0
+  const queue: IComponent[] = [root]
+
+  while (queue.length) {
+    const { length } = queue
+    for (let i = 0; i < length; i++) {
+      const cmp = queue.shift() as IComponent
+      if (expandedCmp[cmp.id]) {
+        cmp.children?.forEach((child) => queue.push(child))
       }
     }
+    depth++
   }
+
   return depth
```

Comment:
Hi @sampullman 

I change the way to calculate tree depth from using a stack to using a queue. There's not much of a difference, but I think it makes it easier to fix the padding issue. I hope this is okay 😬.

Reply:
No problem!